packages/core/core/src/resolveOptions.js

```diff
@@ -38,6 +39,16 @@ export default async function resolveOptions(
     await loadEnv(path.join(rootDir, 'index'));
   }
 
+  let projectRoot = path.dirname(
+    (await resolveConfig(path.join(process.cwd(), 'index'), [
+      'yarn.lock',
+      'package-lock.json',
+      'pnpm-lock.yaml',
+      '.git',
+      '.hg'
+    ])) || path.join(process.cwd(), 'index')
+  );
```

Comment:
Are we concerned about this affecting start up time? Would not be great for people who are not building from a package or git repo, but I'm not sure how often that happens.

Reply:
Also this would maybe not work well if people were using a monorepo of git sub modules and building from within one of those sub modules. Is that something we care to support?

Reply:
> Also this would maybe not work well if people were using a monorepo of git sub modules and building from within one of those sub modules. Is that something we care to support?

This is probably going come come up at some point.

> Are we concerned about this affecting start up time? 

Since this doesn't depend on stuff from earlier in this function, and is async, we could probably run this concurrently with the other async results above with `await Promise.all(...)`